pandas/_libs/window/indexers.pyx

```diff
@@ -10,64 +10,122 @@ from numpy cimport ndarray, int64_t
 # These define start/end indexers to compute offsets
 
 
-class FixedWindowIndexer:
-    """
-    create a fixed length window indexer object
-    that has start & end, that point to offsets in
-    the index object; these are defined based on the win
-    arguments
-
-    Parameters
-    ----------
-    values: ndarray
-        values data array
-    win: int64_t
-        window size
-    index: object
-        index of the values
-    closed: string
-        closed behavior
-    """
-    def __init__(self, ndarray values, int64_t win, object closed, object index=None):
+class BaseIndexer:
+    """Base class for window bounds calculations"""
+
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        **kwargs :
+            keyword argument that will be available when get_window_bounds is called
+        """
+        self.__dict__.update(kwargs)
```

Comment:
What is this doing?

Reply:
Yah can we avoid this pattern?

Reply:
Yeah sorry, it's a bit cryptic.

The motivation is to allow a way for `get_window_bounds` to access an _external_ variable that may not be passed in `rolling` or the aggregation function by making it accessible as a class attribute.

For example:
```
class MyIndexer(BaseIndexer):

    def get_window_bounds(self, ...):
        start = self.external_column + min_periods
        ....

indexer = MyIndexer(external_column=df['external_column'])
``` 

Could make this pattern in the `__init__` more explicit and add validation.